executor/tiflash_test.go

```diff
@@ -93,6 +93,36 @@ func (s *tiflashTestSuite) TestReadPartitionTable(c *C) {
 	tk.MustExec("commit")
 }
 
+func (s *tiflashTestSuite) TestReadUnsigedPK(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("drop table if exists t1")
+	tk.MustExec("create table t(a int unsigned not null primary key, b int not null)")
+	tk.MustExec("alter table t set tiflash replica 1")
+	tb := testGetTableByName(c, tk.Se, "test", "t")
+	err := domain.GetDomain(tk.Se).DDL().UpdateTableReplicaInfo(tk.Se, tb.Meta().ID, true)
+	c.Assert(err, IsNil)
+	tk.MustExec("insert into t values(1,0)")
+	tk.MustExec("insert into t values(2,0)")
+	tk.MustExec("insert into t values(3,0)")
+
+	tk.MustExec("create table t1(a int unsigned not null primary key, b int not null)")
+	tk.MustExec("alter table t1 set tiflash replica 1")
+	tb = testGetTableByName(c, tk.Se, "test", "t1")
+	err = domain.GetDomain(tk.Se).DDL().UpdateTableReplicaInfo(tk.Se, tb.Meta().ID, true)
+	c.Assert(err, IsNil)
+	tk.MustExec("insert into t1 values(1,0)")
+	tk.MustExec("insert into t1 values(2,0)")
+	tk.MustExec("insert into t1 values(3,0)")
+
+	tk.MustExec("set @@session.tidb_isolation_read_engines=\"tiflash\"")
+	tk.MustExec("set @@session.tidb_allow_mpp=ON")
+	tk.MustExec("set @@session.tidb_opt_broadcast_join=ON")
+
```

Comment:
1. add some large int data around maxInt64
2. add more sql to read data, not just count(*)
3. join like -1 = unsigned -1

Reply:
I jibe with the first and last suggestion. But what's the purpose of the 3rd and 4th suggestions ?
Add more sql, but what kinds? Our purpose is to check whether we can deal with unsigned ranges, there are no difference with sql operators. As long as you are using table scan, you are using the same logic to build ranges and dispatch tasks.
I have to admit that testing different senerios with unsigned types is really nesessary, but these tests will target on the TiFlash's runtime. That's not the purpose of TiDB UT involved with mock TiFlash. If you write a mock engine then test the behaviour of the mock engine, that's pointless, even ridiculous.

Reply:
could we add more related test on tiflash, not just on mock tiflash?

Reply:
Of course. I would add the same tests on TiFlash, but at least we have to wait for the tidb's fix before active TiFlash's e2e test, right?

Reply:
if you run some tests in cop or in MPP on tiflash for this PR, and tests are ok, I will LGTM.

Reply:
As the description has mentioned, I have run some tests mannually and made sure that this fix really works well.

Reply:
OK